### PR TITLE
Convert ConfirmButton to functional component

### PR DESCRIPTION
**Pull request by themakshter**
This is to modernise the component where we can use hooks, which results in a smaller function and a smaller bundle size.

### CODE REVIEW

**Comment by themakshter** [comment]
Btw, when you merge my changes, please do a squash & merge and use the PR title + description - I tend to just write random stuff when making commits because I'd rather explain on the PR 😄

File: src/app/components/ConfirmButton/index.tsx [hunk 1]
```diff
@@ -13,115 +14,91 @@ interface IProps {
...
+    setDoing(true);
+    setError(null);
+    onConfirm?.()
       .then(() => {
-        if (this.props.stopSpinnerOnCompletion !== true) {
+        if (stopSpinnerOnCompletion !== true) {
           return;
         }
-        this.setState({
-          show: this.state.show,
```
Comment (themakshter) [inline src/app/components/ConfirmButton/index.tsx:69]: I wasn't sure about this one - I've tried to save the value of the previous state and I think this should be okay. Let me know how I should test this

File: src/app/components/ConfirmButton/index.tsx [hunk 2]
```diff
@@ -13,115 +14,91 @@ interface IProps {
...
-        disabled: true,
-      };
-    }
-    if (this.state.error) {
-      buttonProps = {
-        ...buttonProps,
-        negative: true,
-      };
-      tooltip = this.state.error;
-    }
```
Comment (themakshter) [inline src/app/components/ConfirmButton/index.tsx:102]: I was thinking of moving generating ButtonProps into a separate function, but thought to not further complicate this particular change

File: src/app/components/ConfirmButton/index.tsx (hunk 1 above)
Comment (themakshter) [reply src/app/components/ConfirmButton/index.tsx:69]: Nvm, I've tested it and I don't need to set this. I've removed it.

**Review by drimpact — APPROVED** [review]
LGTM - thanks

**Pull request merged**